output.go

```diff
@@ -0,0 +1,14 @@
+package shaping
+
+import "image"
+
+type Output interface {
+	// Advance returns the distance the Dot has advanced.
+	Advance() int
```

Comment:
What are the units here? I wonder if floating point may be required?

Reply:
Fwiw, x/image/text uses fixed point units (fixed.Int26_6). They're better for portability, and may even be faster in some cases.

Reply:
Good point, this and the Input.TextSize should probably have the same unit. fixed.Int26_6 will offer better performance but IMO is a pain to work with, though there maybe some benefit in adopting the same unit so we can reuse as much of the std lib.